src/WebJobs.Script.WebHost/Diagnostics/MetricsEventManager.cs

```diff
@@ -101,24 +119,34 @@ public void Dispose()
                 GC.SuppressFinalize(this);
             }
 
-           internal void FunctionStarted()
+           internal void FunctionStarted(FunctionStartedEvent startedEvent)
            {
                 totalExecutionCount++;
```

Comment:
Here there are non-thread safe increments, but below you're using a concurrent queue. Does this class need to be threadsafe or not?